### ISSUE TITLE
[Bug]: App freezes on Splash screen after notification open in Android.

### ISSUE THREAD

**Issue by luis901101** [OneSignal/OneSignal-Flutter-SDK#668]
### What happened?

In Android, if app is closed *(I mean user pressed back button until app get closed or simply removed app from recent apps)* when a notification is received and the user taps on it, app starts and freezes in Splash screen.
If app were active in background "not closed", everything works as expected when user taps on notification.

**Note:** 
- App also have configurations for deep links in `AndroidManifest`.
- I have tested with different launch modes for `.MainActivity` in `AndroidManifest` like:  `android:launchMode="singleTop|singleTask|singleInstance"`.

### Steps to reproduce?

```Markdown
1. Using `onesignal_flutter: ^3.5.0`
2. Receiving a push notification with the application closed
3. Tap on notification
4. App starts and freezes in Splash Screen.

## `flutter doctor -v`

[✓] Flutter (Channel stable, 3.7.4, on macOS 13.0 22A380 darwin-arm64, locale en-US)
    • Flutter version 3.7.4 on channel stable at /Users/user/fvm/versions/3.7.4
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b4bce91dd0 (2 weeks ago), 2023-02-21 09:50:50 +0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1

Checking Android licenses is taking an unexpectedly long time...[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/user/Android/SDK
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/user/Android/SDK
    • Java binary at: /Users/user/.sdkman/candidates/java/current/bin/java
    • Java version OpenJDK Runtime Environment Zulu11.56+19-CA (build 11.0.15+10-LTS)
    • All Android licenses accepted.
```

### What did you expect to happen?

I expected app to open normally and receive OpenedHandler as it currently does when it's active in background.

### OneSignal Flutter SDK version

3.5.0

### Which platform(s) are affected?

- [ ] iOS
- [X] Android

### Relevant log output

```Shell
------Following logs are when app is active in background and user taps on notification:------

2023-03-08 12:19:01.242  1490-3920  ActivityTaskManager     system_server                        I  START u0 {flg=0x24000000 cmp=com.myapp/com.onesignal.NotificationOpenedReceiver (has extras)} from uid 10272
2023-03-08 12:19:01.243  1490-3920  ActivityTaskManager     system_server                        W  startActivity called from non-Activity context; forcing Intent.FLAG_ACTIVITY_NEW_TASK for: Intent { flg=0x24800000 cmp=com.myapp/com.onesignal.NotificationOpenedReceiver (has extras) }
2023-03-08 12:19:01.260  1490-1518  ActivityTaskManager     system_server                        I  The Process com.myapp Already Exists in BG. So sending its PID: 32390
2023-03-08 12:19:01.260  2517-2517  NotificationListener    com.android.systemui                 D  onNotificationRemoved: StatusBarNotification(pkg=com.myapp user=UserHandle{0} id=-1931120321 tag=null key=0|com.myapp|-1931120321|null|10272: Notification(channel=fcm_fallback_notification_channel pri=0 contentView=null vibrate=null sound=null tick defaults=0x0 flags=0x10 color=0x00000000 vis=PUBLIC)) reason: 1
2023-03-08 12:19:01.341  1490-3935  ActivityTaskManager     system_server                        I  START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.myapp/.MainActivity} from uid 10272
2023-03-08 12:19:01.449  1490-1617  ActivityTaskManager     system_server                        I  Displayed com.myapp/.MainActivity: +202ms
2023-03-08 12:19:01.475  1490-2493  ClipboardService        system_server                        E  Denying clipboard access to com.myapp, application is not in focus neither is a system service for user 0
2023-03-08 12:19:01.870  2963-2963  GoogleInpu...hodService com...gle.android.inputmethod.latin  I  GoogleInputMethodService.onStartInput():1877 onStartInput(EditorInfo{inputType=0x0(NULL) imeOptions=0x0 privateImeOptions=null actionName=UNSPECIFIED actionLabel=null actionId=0 initialSelStart=-1 initialSelEnd=-1 initialCapsMode=0x0 hintText=null label=null packageName=com.myapp fieldId=-1 fieldName=null extras=null}, false)
2023-03-08 12:19:01.887   791-791   Layer                   surfaceflinger                       E  [Surface(name=AppWindowToken{742f2d7 token=Token{e2bd256 ActivityRecord{1fc9471 u0 com.myapp/.MainActivity t7520}}})/@0x87baad5 - animation-leash#0] No local sync point found
2023-03-08 12:19:02.471  3121-320   Fitness                 com.google.android.gms.persistent    I  OnPackageChangedOperation got intent: Intent { act=android.intent.action.PACKAGE_CHANGED dat=package:com.myapp flg=0x45000010 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.PersistentIntentOperationService (has extras) } [CONTEXT service_id=17 ]
2023-03-08 12:19:02.507  2714-2714  ImsResolver             com.android.phone                    D  maybeAddedImsService, packageName: com.myapp




------Following logs are when app is closed and user taps on notification:------

2023-03-08 12:19:59.073  1490-2642  ActivityTaskManager     system_server                        I  START u0 {flg=0x24000000 cmp=com.myapp/com.onesignal.NotificationOpenedReceiver (has extras)} from uid 10272
2023-03-08 12:19:59.075  1490-2642  ActivityTaskManager     system_server                        W  startActivity called from non-Activity context; forcing Intent.FLAG_ACTIVITY_NEW_TASK for: Intent { flg=0x24800000 cmp=com.myapp/com.onesignal.NotificationOpenedReceiver (has extras) }
2023-03-08 12:19:59.093  1490-2642  ActivityTaskManager     system_server                        I  The Process com.myapp Already Exists in BG. So sending its PID: 729
2023-03-08 12:19:59.128  2517-2517  NotificationListener    com.android.systemui                 D  onNotificationRemoved: StatusBarNotification(pkg=com.myapp user=UserHandle{0} id=-254579642 tag=null key=0|com.myapp|-254579642|null|10272: Notification(channel=fcm_fallback_notification_channel pri=0 contentView=null vibrate=null sound=null tick defaults=0x0 flags=0x10 color=0x00000000 vis=PUBLIC)) reason: 1
2023-03-08 12:19:59.248  1490-10537 ActivityTaskManager     system_server                        I  START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.myapp/.MainActivity} from uid 10272
2023-03-08 12:19:59.277  1490-10537 ActivityTaskManager     system_server                        I  The Process com.myapp Already Exists in BG. So sending its PID: 729
2023-03-08 12:19:59.328   729-729   com.myapp               com.myapp                            W  type=1400 audit(0.0:2235): avc: denied { read } for name="max_map_count" dev="proc" ino=866773 scontext=u:r:untrusted_app:s0:c16,c257,c512,c768 tcontext=u:object_r:proc_max_map_count:s0 tclass=file permissive=0
2023-03-08 12:19:59.358   729-729   com.myapp               com.myapp                            W  type=1400 audit(0.0:2236): avc: denied { search } for name="kgsl-3d0" dev="sysfs" ino=30802 scontext=u:r:untrusted_app:s0:c16,c257,c512,c768 tcontext=u:object_r:sysfs_kgsl:s0 tclass=dir permissive=0
2023-03-08 12:19:59.542   729-729   com.myapp               com.myapp                            W  Accessing hidden method Landroid/view/WindowInsets$Type;->ime()I (blacklist, linking, denied)
2023-03-08 12:19:59.549   729-729   com.myapp               com.myapp                            W  Accessing hidden method Landroid/view/WindowInsets$Type;->ime()I (blacklist, linking, denied)
2023-03-08 12:19:59.769  2963-2963  GoogleInpu...hodService com...gle.android.inputmethod.latin  I  GoogleInputMethodService.onStartInput():1877 onStartInput(EditorInfo{inputType=0x0(NULL) imeOptions=0x0 privateImeOptions=null actionName=UNSPECIFIED actionLabel=null actionId=0 initialSelStart=-1 initialSelEnd=-1 initialCapsMode=0x0 hintText=null label=null packageName=com.myapp fieldId=-1 fieldName=null extras=null}, false)
2023-03-08 12:19:59.774  1490-3947  TelephonyPermissions    system_server                        W  reportAccessDeniedToReadIdentifiers:com.myapp:getSerial:isPreinstalled=false:isPrivApp=false
2023-03-08 12:19:59.787   791-791   Layer                   surfaceflinger                       E  [Surface(name=AppWindowToken{1a3fea2 token=Token{bfbd26d ActivityRecord{7c0ba84 u0 com.myapp/.MainActivity t7523}}})/@0x66d5452 - animation-leash#0] No local sync point found
2023-03-08 12:19:59.813   729-1328  chatty                  com.myapp                            I  uid=10272(com.myapp) 1.ui identical 5 lines
2023-03-08 12:20:00.697  3121-1221  Fitness                 com.google.android.gms.persistent    I  OnPackageChangedOperation got intent: Intent { act=android.intent.action.PACKAGE_CHANGED dat=package:com.myapp flg=0x45000010 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.PersistentIntentOperationService (has extras) } [CONTEXT service_id=17 ]
2023-03-08 12:20:00.731  2714-2714  ImsResolver             com.android.phone                    D  maybeAddedImsService, packageName: com.myapp
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luis901101** [comment]
Any comments on this?

**Comment by anthonycuervo23** [comment]
I have same issue, any solution?

**Comment by luis901101** [comment]
Already fixed in latest ^5.0.0
